### ISSUE TITLE
Cluster Delete throwing error from ring

### ISSUE THREAD

**Issue by MohamedAshiqrh** [heketi/heketi#789]
```
[negroni] Started DELETE /clusters/25bf014d7261274917462770b9bac876
[heketi] ERROR 2017/06/21 17:40:29 /src/github.com/heketi/heketi/apps/glusterfs/allocator_simple.go:142: Unknown cluster id requested: 25bf014d7261274917462770b9bac876
[heketi] INFO 2017/06/21 17:40:29 Deleted cluster [25bf014d7261274917462770b9bac876]
[negroni] Completed 200 OK in 6.751094ms

```

Basically clusterId was not added to ring allocator as it did not have devices. So this check shouldn't throw errors.

**Comment by lpabon** [comment]
:scream: Ahh! that is not good. Do you know a soultion?

**Comment by purpleidea** [comment]
i'm looking into this...

**Comment by purpleidea** [comment]
So after reading code for a while, there are a lot of unknowns...
Because the code has few "docstrings" it's extremely hard to know what's going on, or what the `allocator`'s are for example.
But what's happening is:

When you run a cluster create, it adds an entry into the bolt db, but it *doesn't* add anything into s.rings ... when we delete, it checks that s.rings has an entry and (surprise) finds nothing...

So AFAICT, even within heketi, there are two sources of truth. So if the s.rings cache is needed for something, it's sure not being updated correctly.

Someone with more structural knowledge of what's going on in the code and the core design might need to advice please.

I'd also recommend NACK-ing any patches that don't properly document functions.

**Comment by purpleidea** [comment]
Here's part one of the patch: https://github.com/heketi/heketi/pull/850

While this fixes half of it, I recommend the architect for the project review to make sure there isn't a structural cache-consistency problem with the design. I can imagine this could get quite dangerous if storage and user data is being managed.

**Comment by obnoxxx** [comment]
fixed by PR #850
